smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftSymbolProvider.kt

```diff
@@ -243,25 +249,30 @@ class SwiftSymbolProvider(private val model: Model, val swiftSettings: SwiftSett
             .build()
     }
 
-    private fun numberShape(shape: Shape?, typeName: String, defaultValue: String = "0"): Symbol {
-        if (shape != null && shape.isIntEnumShape()) {
+    private fun numberShape(shape: Shape, typeName: String): Symbol {
+        if (shape.isIntEnumShape()) {
             return createEnumSymbol(shape)
         }
-        return createSymbolBuilder(shape, typeName, "Swift", SwiftDeclaration.STRUCT).putProperty(SymbolProperty.DEFAULT_VALUE_KEY, defaultValue).build()
+        return createSymbolBuilder(shape, typeName, "Swift", SwiftDeclaration.STRUCT).build()
     }
 
     /**
      * Creates a symbol builder for the shape with the given type name in the root namespace.
      */
-    private fun createSymbolBuilder(shape: Shape?, typeName: String, declaration: SwiftDeclaration, boxed: Boolean = false): Symbol.Builder {
+    private fun createSymbolBuilder(
+        shape: Shape,
+        typeName: String,
+        declaration: SwiftDeclaration,
+        boxed: Boolean = false
+    ): Symbol.Builder {
         val builder = Symbol.builder()
             .putProperty("shape", shape)
             .putProperty("decl", declaration.keyword)
             .name(typeName)
         if (boxed) {
             builder.boxed()
         }
-        return builder
+        return handleDefaultValue(shape, builder)
```

Comment:
Before returning the builder, handle any default value it should have in its properties.